app/controllers/crawl_results_controller.rb

```diff
@@ -18,6 +18,10 @@ def index
 
   def show; end
 
+  def source
```

Comment:
The Rails framework has several conventions (such as controller method naming, e.g., `show`) that are best to follow as it helps to make the code more consistent, especially in a team environment. So prefer to have numerous controllers instead of adding non-standard methods:

```ruby
CrawlResultsHTMLController < ApplicationController
  before_action :set_crawl_result

  def source
     render html: @crawl_result.source.html_safe
  end
end
```